app/authentication/session_management.py

```diff
@@ -33,30 +38,103 @@ def get_user_id(self):
         pass
 
 
-class FlaskSessionManager(SessionManagement):
-    def __init__(self):
-        self.logger = logging.getLogger(__name__)
+class DatabaseSessionManager(SessionManagement):
 
-    def store_user_id(self, jwt):
+    def store_user_id(self, user_id):
+        logger.debug("DatabaseSessionManager store_user_id() - session %s", session)
+        if EQ_SESSION_ID not in session:
+            eq_session_id = self.create_session_id()
+            logger.debug("Created new eq session id %s", eq_session_id)
+            session[EQ_SESSION_ID] = eq_session_id
+            session.permanent = True
+            eq_session = EQSession(eq_session_id, user_id)
+            logger.debug("Constructed EQ Session object %s", eq_session)
+        else:
+            eq_session_id = session[EQ_SESSION_ID]
+            logger.debug("Found eq_session_id %s in session", eq_session_id)
+            eq_session = self._get_object(eq_session_id)
+            logger.debug("Loaded object eq session %s", eq_session)
+        logger.debug("About to commit to database")
+        db_session.add(eq_session)
+        db_session.commit()
+        logger.debug("Committed")
+
+    def has_user_id(self):
+        logger.debug("DatabaseSessionManager has_user_id() - session %s", session)
+        if EQ_SESSION_ID in session:
+            eq_session_id = session[EQ_SESSION_ID]
+
+            count = self.run_count(eq_session_id)
+            logger.debug("Number of entries for eq session id %s is %s", eq_session_id, count)
+            return count > 0
+
+    def remove_user_id(self):
+        logger.debug("DatabaseSessionManager remove_user_id() - session %s", session)
+        if EQ_SESSION_ID in session:
+            eq_session_id = session[EQ_SESSION_ID]
+            eq_session = self._get_object(eq_session_id)
+            logger.debug("About to delete entry from eq_session table %s", eq_session)
+            db_session.delete(eq_session)
+            db_session.commit()
+        else:
+            logger.warning("No eq session id exists")
+
+    def get_user_id(self):
+        logger.debug("DatabaseSessionManager get_user_id() - session %s", session)
+        if EQ_SESSION_ID in session:
+            eq_session_id = session[EQ_SESSION_ID]
+            eq_session = self._get_object(eq_session_id)
+            return eq_session.user_id
+        else:
+            return None
+
+    def create_session_id(self):
+        while True:
```

Comment:
Since this should probably rarely occur, if at all; if it does happen after multiple retires it would suggest something more serious is wrong (log error/throw exception?)